From b8c0f1844909de744ce58075aa671893cdb5d78e Mon Sep 17 00:00:00 2001
From: Ronald Claveau <linux-kernel-dev@aliel.fr>
Date: Fri, 3 Apr 2026 12:28:34 +0200
Subject: [PATCH 80/91] arm64: dts: amlogic: t7: khadas-vim4: Enable USB

Enable USB combo controller and set the PHY regulators
for the Khadas VIM4 board.

Signed-off-by: Ronald Claveau <linux-kernel-dev@aliel.fr>
Signed-off-by: Nick Xie <nick@khadas.com>
---
 .../amlogic/amlogic-t7-a311d2-khadas-vim4.dts    | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts b/arch/arm64/boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts
index ba9219073dd0..e1468349a52a 100644
--- a/arch/arm64/boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts
+++ b/arch/arm64/boot/dts/amlogic/amlogic-t7-a311d2-khadas-vim4.dts
@@ -368,3 +368,19 @@ map {
 		};
 	};
 };
+
+&usb_comb {
+	status = "okay";
+};
+
+&usb2_phy0 {
+	phy-supply = <&vcc5v0_usb>;
+};
+
+&usb2_phy1 {
+	phy-supply = <&vcc5v0_usb>;
+};
+
+&usb3_phy {
+	phy-supply = <&vddao_1v8>;
+};
-- 
2.53.0

